src/components/ConditionSelector/Condition.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import './Condition.scss';
+import * as classNames from 'classnames';
+import { ConditionRowState, ConditionDefinition, PropertyTypeEnum } from './';
+import { Dropdown, DropdownType, IDropdownOption } from '../Dropdown';
+import { Button } from '../Button';
+import { TextField } from '../TextField';
+import { DateTimeDropdownPicker } from '../DateTimeDropdownPicker/DateTimeDropdownPicker';
+import { DragSource } from 'react-dnd';
+
+const conditionSource = {
+    beginDrag(props) {
+        return {};
+    },
+    endDrag(props, monitor, component) {
+        const dropTarget = monitor.getDropResult();
+        if (Boolean(props) && Boolean(dropTarget) && component.props.conditionDragged) {
+            component.props.conditionDragged(props, dropTarget);
+        }
+        return {};
+    }
+};
+
+function collectSource(connect, monitor) {
+    return {
+        connectDragSource: connect.dragSource(),
+        connectDragPreview: connect.dragPreview(),
+        isDragging: monitor.isDragging()
+    };
+}
+
+@DragSource('condition', conditionSource, collectSource)
+export class Condition extends React.PureComponent <ConditionDefinition, any> {
+    public static defaultProps = {
+        hasIndent: false,
+        indentSize: 30,
+        isHardcodedValue: false
+    };
+
+    compareConditionChanged = (option: IDropdownOption, index?: number) => {
+        const state: ConditionRowState  = {
+            propertyId: this.props.id,
+            selectedCompareConditionKey: option.key
+        };
+        if (this.props.compareConditionChanged) {
+           this.props.compareConditionChanged(state);
+        }
+    }
+    
+    renderItemEditor = (propertyType: PropertyTypeEnum,  additionalData: any) => {
+        switch (propertyType) {
+            case PropertyTypeEnum.String: {
```

Comment:
It would be better if these cases would be either functions that return JSX elements or even better separate components (this seems a perfect use case for React Functional Components https://reactjs.org/docs/components-and-props.html )